File: 12-doubly-linked-list/01-insert-at-beginning-of-dll/01-insert-at-beginning-of-dll.py
import sys
import os
sys.path.append(os.getcwd())
from PythonDataStructures.DoublyLinkedList.ListNode import ListNode

def insert_at_beginning_of_doubly_linked_list(head, x):
  node = ListNode(x, None, head)
  if head:
    head.prev = node
  return node


head = ListNode.create([10, 15, 20])
res = insert_at_beginning_of_doubly_linked_list(head, 5)
print(res)

File: 12-doubly-linked-list/01-insert-at-beginning-of-dll/01-insert-at-beginning-of-dll.rb
require_relative '../../RubyDataStructure/DoublyLinkedList/ListNode.rb'

def insert_at_beginning_of_doubly_linked_list(head, key)
  return ListNode.new(key, head)
end

l1 = ListNode.new(10)
l2 = ListNode.new(20, nil, l1)
l1.next = l2

res = insert_at_beginning_of_doubly_linked_list l1, 5
while res
  print "#{res.val} "
  res = res.next
end